tokio-executor/src/global.rs

```diff
@@ -40,59 +40,57 @@ impl DefaultExecutor {
             _dummy: (),
         }
     }
+
+    #[inline]
+    fn with_current<F: FnOnce(&mut Executor) -> R, R>(f: F) -> Option<R> {
+        EXECUTOR.with(|current_executor| {
+            match current_executor.replace(State::Active) {
+                State::Ready(executor_ptr) => {
+                    let executor = unsafe { &mut *executor_ptr };
+                    let result = f(executor);
```

Comment:
There is an issue w/ panics here. If `f()` panics, the state stays "active".

The solution is to use something like this: https://github.com/srijs/tokio/blob/0348be3f79e7164941554d7aaf1c500f515a50aa/tokio-executor/src/global.rs#L174

Reply:
Sorry for not noticing earlier.

Reply:
No problem!

I noticed that this could happen, but since it doesn't lead to any unsafe behaviour (contrary to the `with_default` example you posted), I wasn't sure if it was going to be a problem.

I'll fix it up later today.